### PR TITLE
Close #2038: Dumping Ehcache state to state dump

### LINKED ISSUE #2038: Add Ehcache state information inside server state dumps

### PR DESCRIPTION
**Pull request by mathieucarbou**
I'll add a test on tuesday, but let me know if you have some comments.... Thanks!

Note: I have opened:

- https://github.com/Terracotta-OSS/terracotta-apis/issues/240
- https://github.com/Terracotta-OSS/terracotta-apis/issues/241
- https://github.com/Terracotta-OSS/terracotta-apis/issues/242

### CODE REVIEW

File: clustered/server/src/main/java/org/ehcache/clustered/lock/server/VoltronReadWriteLockActiveEntity.java [hunk 1]
```diff
@@ -57,6 +60,25 @@ public LockTransition invoke(ClientDescriptor client, LockOperation message) {
...
+    dump.addState("exclusiveHolder", String.valueOf(exclusiveHolder));
+    {
+      StateDumpCollector releaseListenersDump = dump.subStateDumpCollector("releaseListeners");
+      int idx = 0;
+      for (ClientDescriptor releaseListener : new HashSet<>(this.releaseListeners)) {
+        releaseListenersDump.addState(String.valueOf(idx++), releaseListener.toString());
+      }
+    }
+    {
+      StateDumpCollector sharedHolders = dump.subStateDumpCollector("sharedHolders");
```
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/lock/server/VoltronReadWriteLockActiveEntity.java]: Shared holders should be dumped at the same level as exclusive holder. And actually it is an either or situation.

**Comment by mathieucarbou** [comment]
@ljacomet @anthonydahanne : PR updated and I added a little test that we could improve once the state dump format will be better...

**Comment by mathieucarbou** [comment]
I have cleaned up the diff (imports reordering) and dumped pools also from state service

File: clustered/integration-test/src/test/java/org/ehcache/clustered/management/DiagnosticTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        .heap(10, EntryUnit.ENTRIES)
+        .offheap(1, MemoryUnit.MB)
+        .with(clusteredDedicated("primary-server-resource", 2, MemoryUnit.MB)))
+      .build());
+
+    Properties properties = new Properties();
+    properties.setProperty(ConnectionPropertyNames.CONNECTION_TIMEOUT, String.valueOf("5000"));
+    properties.setProperty(ConnectionPropertyNames.CONNECTION_NAME, "diagnostic");
+    properties.setProperty(PROP_REQUEST_TIMEOUT, "5000");
+    properties.setProperty(PROP_REQUEST_TIMEOUTMESSAGE, "5000");
```
Comment (anthonydahanne) [inline clustered/integration-test/src/test/java/org/ehcache/clustered/management/DiagnosticTest.java]: 5000 ? that's a weird message
not even sure when / how it's gonna be displayed

**Comment by mathieucarbou** [comment]
seems to be aborted....

test this please

**Comment by mathieucarbou** [comment]
test this please

**Pull request merged**